src/infrastructure/validation/logger/FileLogger.java

```diff
@@ -0,0 +1,108 @@
+package infrastructure.validation.logger;
+
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.io.IOException;
+
+/**
+ * FileLogger class that is part of loggerManager,
+ * and used for logging to the a file (the log file)
+ * Implements ILogger interface
+ * 
+ * @author Navaneeth M Nambiar
+ */
+public class FileLogger implements ILogger {
+
+	/** stores the format specified for the time-stamp */
+	private DateTimeFormatter timeStampFormat;
+		
+	/** string that holds the location of the log file */
+	private static String logFile;
+	
+	/**
+	 *  constructor for FileLogger
+	 *  protected type since it needs to be only invoked by LoggerManager class
+	 *  @see logger.LoggerManager
+	 */
+	protected FileLogger() {
+		
+		// sets DateTime format as per the spec
+		timeStampFormat = DateTimeFormatter.ofPattern("yyyy:MM:dd HH:mm:ss");
+		
```

Comment:
the filelogger and console are not given as static. If they are not ensuring single instance, then the data members probably should be referenced with 'this.timeStampFormat'

Reply:
I think sir asked to ensure single instance too. I didn't particularly hear what he suggested. Could you update me?

Reply:
single instance is done at loggerFactory level by the getLoggerInstance() method
constructors for filelogger and consolelogger are protected. So they cant be called externally